File: frontend-app/src/app/types/workoutDTO.ts
export interface ExerciseDTO {
  id: string;
  name: string;
  sets: number;
  reps: number;
  weight?: number;
  muscleGroups?: string[];
  restTimeSec?: number;
}

export interface WorkoutDTO {
  id: string;
  title: string;
  scheduledAt: string;
  completedAt?: string | null;
  exercises: ExerciseDTO[];
  muscleGroup: string;
  notes?: string;
}

File: backend-app/apiModule/ApiModule/WebApplication1/Api/Contracts/Workout/WorkoutDto.cs
namespace ApiModule.Api.Contracts.Workout;

public record WorkoutDto(
    Guid Id,
    string Title,
    DateTime ScheduledAt,
    DateTime? CompletedAt,
    string? PerceivedLoad,
    string[] MuscleGroups,
    List<ExerciseDto> Exercises
);
